akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -344,11 +345,24 @@ private[akka] class PersistentShard(
     case EntityStopped(id)                 ⇒ state = state.copy(state.entities - id)
     case SnapshotOffer(_, snapshot: State) ⇒ state = snapshot
     case RecoveryCompleted ⇒
-      state.entities foreach getEntity
+      restartRememberedEntities()
       super.initialized()
       log.debug("Shard recovery completed {}", shardId)
   }
 
+  private def restartRememberedEntities(): Unit = {
+    if (entityRecoveryRateInterval == Duration.Zero) {
+      state.entities foreach getEntity
+    } else {
+      state.entities.foldLeft(entityRecoveryRateInterval) {
+        case (interval, entityId) ⇒
+          import context.dispatcher
+          context.system.scheduler.scheduleOnce(interval, self, RestartEntity(entityId))
```

Comment:
Hm, I would probably like to see batches here. Maybe it makes more sense to provide a rate instead of a period?

NB: With streams it would be as easy as
```scala
Source(state.entities).throttle(initialBurst, entityPerSecond).runForeach(self ! RestartEntity(_))
```
Unfortunately this project cannot rely on streams.